verification/verify.go

```diff
@@ -103,7 +112,29 @@ func (sv *SegmentVerifier) Verify(params *Params) (*Params, error) {
 		return nil, nil
 	}
 
-	// TODO sig checking; extract from broadcast.go
+	if params.Results != nil {
+		segHashes := make([][]byte, len(params.Results.Segments))
+		for i, segment := range params.Results.Segments {
+
+			data, err := drivers.GetSegmentData(segment.Url)
+			if err != nil {
+				glog.Errorf("%v error with segment: %v (URL: %v)", monitor.SegmentTranscodeErrorDownload, err, segment.Url)
+				break
+			}
+			segHashes[i] = crypto.Keccak256(data)
+		}
+
+		// Might not have seg hashes if results are directly uploaded to the broadcaster's OS
+		// TODO: Consider downloading the results to generate seg hashes if results are directly uploaded to the broadcaster's OS
+		if len(segHashes) != len(params.Results.Segments) &&
```

Comment:
Let's add tests for this new logic.

Reply:
done in 42fc12235cf02880eb2efde14c92fffde1f85174